pyk/src/tests/unit/test_kcfg.py

```diff
@@ -849,3 +850,69 @@ def test_no_cell_rewrite_to_dots() -> None:
 
     result = no_cell_rewrite_to_dots(term)
     assert result == term
+
+
+CREATE_SPLIT_BY_NODES_TEST_DATA: Final = (
+    (CTerm.bottom(), [CTerm.top(), CTerm.bottom()], None),
+    (CTerm.top(), [CTerm.top(), CTerm.bottom()], None),
+    (
+        CTerm.top(),
+        [CTerm(config('X')), CTerm(config('Y'))],
+        None,  # todo: support split from top, because top means anything can be matched
+    ),
+    (
+        CTerm.top(),
+        [CTerm.top(), CTerm.top()],
+        KCFG.Split(
+            KCFG.Node(1, CTerm.top()), [(KCFG.Node(2, CTerm.top()), CSubst()), (KCFG.Node(3, CTerm.top()), CSubst())]
+        ),
+    ),
+    (
+        CTerm(config('X')),
+        [CTerm(config('X')), CTerm(config('Y')), CTerm(config('Z'))],
+        KCFG.Split(
+            KCFG.Node(1, CTerm(config('X'))),
+            [
+                (KCFG.Node(2, CTerm(config('X'))), CSubst(Subst({'X': KVariable('X')}))),
+                (KCFG.Node(3, CTerm(config('Y'))), CSubst(Subst({'X': KVariable('Y')}))),
+                (KCFG.Node(4, CTerm(config('Z'))), CSubst(Subst({'X': KVariable('Z')}))),
+            ],
+        ),
+    ),
```

Comment:
This one also violates that the targets are mutually exclusive.

Reply:
Let's put a comment that it violates that condition, or update the test to not violate that condition (by having constraints on the target nodes that make them mutually exclusive), eg:

```
[CTerm(config('X'), [gt('X', 3), le('X', 10)]), CTerm(config('Y'), [le('X', 3)]), CTerm(config('Z'), [gt('X', 10)])])],
```